cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -234,6 +234,8 @@ public DatasetAdmin call() throws Exception {
         });
       } catch (Exception e) {
         Throwables.propagateIfInstanceOf(e, IOException.class);
+        Throwables.propagateIfInstanceOf(e, DatasetManagementException.class);
```

Comment:
why is this needed? I can see where NotFoundException is thrown, and seems like it will fix the failure. Not sure about this one.